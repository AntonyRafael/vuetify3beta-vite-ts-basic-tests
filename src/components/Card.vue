<template>
  <v-hover v-slot="{ isHovering, props }">
    <v-card class="pa-2 mb-2" v-bind="props" :elevation="isHovering ? 24 : 8">
      <v-list-item lines="three">
        <v-list-item-content>
          <p class="text-overline mb-4" :title-test="title">
            {{ title }}
          </p>
          <v-list-item-title class="text-h4 mb-1" :stats-test="stats">
            {{ stats }}
          </v-list-item-title>
          <v-list-item-subtitle :content-test="content">
            {{ content }}
          </v-list-item-subtitle>
        </v-list-item-content>

        <v-spacer />

        <template v-slot:append>
          <v-list-item-avatar end class="mt-4">
            <v-icon :icon="icon" size="x-large" :iconName-test="icon" />
          </v-list-item-avatar>
        </template>
      </v-list-item>

      <v-card-actions>
        <v-btn
          :to="pathName"
          color="dark"
          variant="text"
          tabindex="0"
          :buttonName-test="buttonName"
        >
          {{ buttonName }}
        </v-btn>
      </v-card-actions>
    </v-card>
  </v-hover>
</template>

<script lang="ts">
import { defineComponent } from "vue";
export default defineComponent({
  props: {
    id: {
      type: Number,
      default: 0,
    },
    title: {
      type: String,
      default: "",
    },
    content: {
      type: String,
      default: "",
    },
    icon: {
      type: String,
      default: "",
    },
    buttonName: {
      type: String,
      default: "",
    },
    pathName: {
      type: String,
      default: "",
    },
    stats: {
      type: Number,
      default: 0,
    },
  },
});
</script>
