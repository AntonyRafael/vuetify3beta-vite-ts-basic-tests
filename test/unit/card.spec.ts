import { createVuetify } from "vuetify";
import { mount } from "@vue/test-utils";
import { beforeEach, describe, expect, it } from "vitest";
import Card from "../../src/components/Card.vue";

describe("Card", () => {
  let wrapper;

  beforeEach(() => {
    const vuetify = createVuetify();

    wrapper = mount(Card, {
      props: {
        id: 1,
        title: "Title Card 1",
        content: "Content Card 1",
        icon: "mdi-account-group",
        stats: 15,
        buttonName: "Button Test 1",
        pathName: "/",
      },
      global: {
        plugins: [vuetify],
      },
    });
  });

  it("Is a Vue instance", () => {
    expect(wrapper).toBeTruthy();
  });
  it("Renders the component", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Renders the template with data", () => {
    expect(wrapper.vm.id).toBe(1);
    expect(wrapper.vm.title).toBe("Title Card 1");
    expect(wrapper.vm.content).toBe("Content Card 1");
    expect(wrapper.vm.icon).toBe("mdi-account-group");
    expect(wrapper.vm.stats).toBe(15);
    expect(wrapper.vm.buttonName).toBe("Button Test 1");
    expect(wrapper.vm.pathName).toBe("/");
  });

  it("Renders the title with data", () => {
    const title = wrapper.find('[title-test="Title Card 1"]');
    expect(title.exists()).toBe(true);
    expect(title.text()).toBe("Title Card 1");
  });

  it("Renders the stats with data", () => {
    const stats = wrapper.find('[stats-test]');
    expect(stats.exists()).toBe(true);
    expect(stats.text()).toBe("15");
  });

  it("Renders the content with data", () => {
    const content = wrapper.find('[content-test]');
    expect(content.exists()).toBe(true);
    expect(content.text()).toBe("Content Card 1");
  });

  it("Renders the icon name with data", () => {
    const icon = wrapper.find('[iconName-test="mdi-account-group"]');
    expect(icon.exists()).toBe(true);
  });

  it("Renders the button name with data", () => {
    const buttonName = wrapper.find("[buttonName-test]");
    expect(buttonName.exists()).toBe(true);
    expect(buttonName.text()).toBe("Button Test 1");
  });
});
