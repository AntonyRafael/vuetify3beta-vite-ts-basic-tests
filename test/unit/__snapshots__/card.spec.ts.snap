// Vitest Snapshot v1

exports[`Card > Renders the component 1`] = `
"<div class=\\"v-card v-theme--light v-card--density-default elevation-8 v-card--variant-contained pa-2 mb-2\\">
  <!---->
  <div class=\\"v-card__underlay\\"></div>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <div class=\\"v-list-item v-theme--light v-list-item--density-default v-list-item--three-line v-list-item--variant-text\\">
    <!---->
    <div class=\\"v-list-item__underlay\\"></div>
    <!---->
    <!---->
    <v-list-item-content>
      <p class=\\"text-overline mb-4\\" title-test=\\"Title Card 1\\">Title Card 1</p>
      <div class=\\"v-list-item-title text-h4 mb-1\\" stats-test=\\"15\\">15</div>
      <div class=\\"v-list-item-subtitle\\" content-test=\\"Content Card 1\\">Content Card 1</div>
    </v-list-item-content>
    <div class=\\"flex-grow-1\\"></div>
    <div class=\\"v-avatar v-avatar--end v-avatar--density-default v-avatar--size-default v-list-item-avatar v-list-item-avatar--end mt-4\\"><i class=\\"mdi-account-group mdi v-icon notranslate v-icon--size-x-large v-theme--light\\" aria-hidden=\\"true\\" iconname-test=\\"mdi-account-group\\"></i></div>
    <!---->
    <!---->
  </div>
  <div class=\\"v-card-actions\\"><a class=\\"v-btn v-theme--light text-dark v-btn--density-default v-btn--size-default v-btn--variant-text\\" tabindex=\\"0\\" buttonname-test=\\"Button Test 1\\">
      <div class=\\"v-btn__overlay\\"></div>
      <div class=\\"v-btn__underlay\\"></div>
      <!---->
      <div class=\\"v-btn__content\\" data-no-activator=\\"\\">Button Test 1</div>
      <!---->
    </a></div>
  <!---->
</div>"
`;
